src/Lib.hs

```diff
@@ -23,6 +24,7 @@ data Block = Block { index        :: Int
                    , previousHash :: String
                    , timestamp    :: Int
                    , blockData    :: String
+                   , nonce        :: Int
```

Comment:
You mentioned the edge case where this gets too big, which is a great point. It's probably fine for now, but let's at least add a TODO here to address it. I think ideally we'd have a reasonable max on the difficulty if/when we make it configurable. And I think that bumping timestamp and resetting the nonce if/when it overflows is a fine solution too.